plugins/vocabbuilder.koplugin/main.lua

```diff
@@ -0,0 +1,1178 @@
+    self.title = title
+    self:free()
+    self:init()
+    UIManager:setDirty("all", "ui")
+end
+
+function WordInfoDialog:onShow()
+    UIManager:setDirty(self, function()
+        return "flashui", self[1][1].dimen
+    end)
+end
+
+function WordInfoDialog:onCloseWidget()
+    UIManager:setDirty(nil, function()
+        return "ui", self[1][1].dimen
+    end)
+end
+
+function WordInfoDialog:onTapClose()
+    UIManager:close(self)
+    if self.tap_close_callback then
+        self.tap_close_callback()
+    end
+    return true
+end
+
+function WordInfoDialog:onClose()
+    self:onTapClose()
+    return true
+end
+
+function WordInfoDialog:paintTo(...)
+    InputContainer.paintTo(self, ...)
+    self.dimen = self[1][1].dimen -- FrameContainer
+end
+
+
+
+-- values useful for item cells
+local review_button_width = Screen:scaleBySize(95)
+local ellipsis_button_width = Screen:scaleBySize(34)
+local star_width = Screen:scaleBySize(25)
+
+local point_widget = TextWidget:new{
+    text = " • ",
+    bold = true,
+    face = Font:getFace("cfont", 24),
+    fgcolor = dim_color
+}
+
+--[[--
+Individual word item widget
+--]]--
+local VocabItemWidget = InputContainer:new{
+    face = Font:getFace("smallinfofont"),
+    width = nil,
+    height = nil,
+    show_parent = nil,
+    item = nil,
+    forgot_button = nil,
+    got_it_button = nil,
+    more_button = nil,
+    layout = nil
+}
+--[[--
+    item: {
+        checked_func: Block,
+        review_count: interger,
+        word: Text
+        book_title: TEXT
+        create_time: Integer
+        review_time: Integer
+        due_time: Integer,
+        got_it_callback: function
+        remove_callback: function
+        is_dim: BOOL
+    }
+--]]--
+
+local point_widget_height = point_widget:getSize().h
+local point_widget_width = point_widget:getSize().w
+local word_height = TextWidget:new{text = " ", face = word_face}:getSize().h
+local subtitle_height = TextWidget:new{text = " ", face = subtitle_face}:getSize().h
+
+
+function VocabItemWidget:init()
+    self.layout = {}
+    self.dimen = Geom:new{w = self.width, h = self.height}
+    self.ges_events.Tap = {
+        GestureRange:new{
+            ges = "tap",
+            range = self.dimen,
+        }
+    }
+    self.ges_events.Hold = {
+        GestureRange:new{
+            ges = "hold",
+            range = self.dimen,
+        }
+    }
+    self.v_spacer = VerticalSpan:new{width = (self.height - word_height - subtitle_height)/2}
+    self.point_v_spacer = VerticalSpan:new{width = (self.v_spacer.width + word_height/2) - point_widget_height/2 }
+    self.margin_span = HorizontalSpan:new{ width = Size.padding.large }
+    self:initItemWidget()
+end
+
+function VocabItemWidget:initItemWidget()
+    for i = 1, #self.layout do self.layout[i] = nil end
+
+    local word_widget = Button:new{
+        text = self .item.word,
+        bordersize = 0,
+        callback = function() self:onTap() end
+    }
+    if self .item.is_dim then
```

Comment:
Surprised Lua doesn't complain about this space between `left` and its following dot.
But please remove this space (here and in many places below), so it reads naturally.

Reply:
OK will do. I forgot to remove it, was introduced with a replace all.